### PR TITLE
Rewrite beautifyFileSize to use SI (base-1000) units

### PR DESCRIPTION
**Pull request by lakshyajn**
To address the issue under `kiwix-tools` , modified the function to display human-readable size formats.
It can now be used in `kiwix-tools` to display sizes nicely.

Also added new unit tests as per the changes

### CODE REVIEW

**Comment by kelson42** [comment]
@veloman-yunkan I would propose to rename to `beaitifyMemorySize()`, keep `beautifyFileSize` as a wrapper (with deprecation warning and remove next major release). What do you think?

File: src/tools/stringTools.cpp [hunk 1]
```diff
@@ -126,16 +126,39 @@ std::string kiwix::beautifyInteger(uint64_t number)
...
+    999995000000000ULL,      // PB
+    999995000000000000ULL    // EB
+  };
+  static_assert(sizeof(thresholds) / sizeof(thresholds[0]) == unitCount,
+                "thresholds and units arrays must have the same length");
+
+  size_t unit = 0;
+  while (unit + 1 < unitCount && number >= thresholds[unit + 1]) {
+    ++unit;
+  }
```
Comment (veloman-yunkan) [inline src/tools/stringTools.cpp]: The problem that this code is trying to address is too unimportant (this function is not used to control a space telescope; rounding that is slightly off for values falling right on the rounding boundary for the case of exact arithmetic can be easily tolerated). Therefore I consider this solution as overengineered and too cumbersome. I'd rather stick to the original simpler solution and instead adapt the unit tests to allow for either result for an edge case value. With a properly defined helper macro `EXPECT_ONE_OF()` the check can be written as:

```c++
EXPECT_ONE_OF(beautifyFileSize(999995000ULL), ("999.99 GB", "1.00 GB")  );
```

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan I would propose to rename to `beaitifyMemorySize()`, keep `beautifyFileSize` as a wrapper (with deprecation warning and remove next major release). What do you think?

Not worth the effort. In case you choose to do it, at least don't let the typo to sneak in the public API :)

**Comment by lakshyajn** [comment]
@veloman-yunkan  kindly have a look, I added the macro and replaced that "rocket engineering optimisation" 😅 code with almost as the previous one.

**Comment by veloman-yunkan** [comment]
@lakshyajn For some reason, the diff is currently unavailable, but I looked at your changes via [the diff with the previous version](https://github.com/kiwix/libkiwix/compare/ac47877874d6046f24f3955fca844bf8c9e9d980..16124ae39277d943cbb026d46f271eda393fbf71).

Can't you incorporate the correction step into the main unit detection loop by changing the condition to test the rounded value? Note that we don't mind about the performance of this function.

BTW, in order to make the code more readable/clear I would introduce a small helper function `double roundToTwoDecimalPlaces(double)` in an unnamed namespace above.

**Comment by lakshyajn** [comment]
> Can't you incorporate the correction step into the main unit detection loop by changing the condition to test the rounded value?

Yeah sorry about that, it made the function more simpler for checking.
Thanks :)

**Review by veloman-yunkan — APPROVED** [review]
LGTM
